### PR TITLE
Policy gateway type change handling.

### PR DESCRIPTION
**Pull request by slahirucd7**
$subject.

### CODE REVIEW

File: portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx [hunk 1]
```diff
@@ -157,8 +163,8 @@ const GatewaySelector: FC<GatewaySelectorProps> = ({
                                         control={
                                             <Radio
                                                 color='primary'
-                                                disabled
-                                                onChange={handleApprovedGatewayChange}
+                                                disabled={isRestricted(['apim:api_create'], apiFromContext)}
+                                                onChange={() => handleApprovedGatewayChange(true)}
                                             />
                                         }
                                         label='Choreo Connect'
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx:170]: Line 177: Change Typography type to h3/h4 or an appropriate one

File: portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx [hunk 2]
```diff
@@ -145,7 +151,7 @@ const GatewaySelector: FC<GatewaySelectorProps> = ({
                                                     apiFromContext,
                                                 )}
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx:152]: Line 138 aria attribute should be handled in [this way](https://v4.mui.com/components/radio-buttons/#accessibility)

File: portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx [hunk 3]
```diff
@@ -145,7 +151,7 @@ const GatewaySelector: FC<GatewaySelectorProps> = ({
                                                     apiFromContext,
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx:151]: Lets use `RadioGroup` onChange event https://v4.mui.com/api/radio-group/#radiogroup-api

File: portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx [hunk 4]
```diff
@@ -197,7 +203,7 @@ const GatewaySelector: FC<GatewaySelectorProps> = ({
                                         />
                                     </Button>
                                     <Button
-                                        onClick={saveAfterGatewayChange}
+                                        onClick={() => { saveAfterGatewayChange() }}
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx:206]: lets remove the redundent lambda function

File: portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx (hunk 2 above)
Comment (slahirucd7) [reply portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx:155]: Fixed with: https://github.com/wso2/apim-apps/pull/211/commits/d6d0f77285306bbc17f0e0ccc35006a166d5429f

File: portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx (hunk 3 above)
Comment (slahirucd7) [reply portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx:154]: Fixed with: https://github.com/wso2/apim-apps/pull/211/commits/97db4049a34281272f86383e7fc0372b84f8f83c

File: portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx (hunk 1 above)
Comment (slahirucd7) [reply portals/publisher/source/src/app/components/Apis/Details/Policies/GatewaySelector.tsx:175]: Fixed with: https://github.com/wso2/apim-apps/pull/211/commits/97db4049a34281272f86383e7fc0372b84f8f83c

**Review by Krishanx92 — APPROVED** [review]

**Pull request merged**